app/controllers/courses_controller.rb

```diff
@@ -172,6 +255,454 @@ def userLookup
 
   end
 
+  action_auth_level :users, :instructor
+  def users
+    if(params[:search]) then
+      # left over from when AJAX was used to find users on the admin users list
+      @cuds = @course.course_user_data.joins(:user).order('users.email ASC').where(CourseUserDatum.conditions_by_like(params[:search]))
+    else
+      @cuds = @course.course_user_data.joins(:user).order('users.email ASC')
+    end
+
+  end
+
+
+  action_auth_level :sudo, :instructor
+  def sudo
+    session[:sudo] = nil
+    redirect_to course_course_user_datum_sudo_path(course_user_datum_id: @cud.id)
+  end
+
+
+  action_auth_level :reload, :instructor
+  def reload
+    if @course.reload_course_config then
+      flash[:success] = "Success: Course config file reloaded!"
+      redirect_to action: :show and return
+    else
+      render and return
+    end
+  end
+
+
+  # Upload a CSV roster and import the users into the course
+  # Colors are associated to each row of CUD after roster is processed:
+  #   green - User doesn't exist in the course, and is going to be added
+  #   red - User is going to be dropped from the course
+  #   black - User exists in the course
+  action_auth_level :uploadRoster, :instructor
+  def uploadRoster
+    if request.post? then
```

Comment:
Check the routes, but this check may be unnecessary (somewhat out of scope of this pr).